Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -761,10 +761,18 @@ def network_shutdown(self):
 
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
+
+        # if pstate can't be found
+        if not pstate.get('state', 'engineresumedata'):
```

Comment:
Shouldn't this be refreshed every time the method is called? 
Otherwise we will be using stale information from that point on.

If we did this then everything on this function could be moved to `on_save_resume_data_alert` which would make it a bit nicer.